tracer/src/Datadog.Tracer.Native/tracer_tokens.cpp

```diff
@@ -351,7 +421,26 @@ HRESULT TracerTokens::WriteBeginMethod(void* rewriterWrapperPtr, mdTypeRef integ
     {
         const auto [elementType, argTypeFlags] = methodArguments[i].GetElementTypeAndFlags();
 
-        if (enable_by_ref_instrumentation && (argTypeFlags & TypeFlagByRef))
+        bool isByRefLike = false;
+        if (FAILED(IsTypeByRefLike(_profiler_info, *module_metadata, methodArguments[i], GetCorLibAssemblyRef(), isByRefLike)))
+        {
+            isByRefLike = false;
+        }
+
+        if (enable_by_ref_instrumentation && isByRefLike)
+        {
+            unsigned calltargetRefStructTypeBuffer;
+            ULONG calltargetRefStructTypeSize = CorSigCompressToken(callTargetRefStructTypeRef, &calltargetRefStructTypeBuffer);
+
+            auto argSignature = new COR_SIGNATURE[calltargetRefStructTypeSize + 1];
```

Comment:
I don't see where you're freeing it

Reply:
fixed here: [ebb9f20](https://github.com/DataDog/dd-trace-dotnet/pull/5442/commits/ebb9f201fa90c635bc411fe019d802a2ddcde7d4)